pkg/domain/node.go

```diff
@@ -137,6 +138,14 @@ type PeekableDependentProperty struct {
 	ValueKey    string `json:"value_key"`
 }
 
+type PeekablePaginationConfig struct {
+	DefaultLimit            int      `json:"default_limit,omitempty"`
+	MaxLimit                int      `json:"max_limit,omitempty"`
+	SupportsOrderBy         bool     `json:"supports_order_by,omitempty"`
+	OrderByOptions          []string `json:"order_by_options,omitempty"`
+	SupportsIncludeArchived bool     `json:"supports_include_archived,omitempty"`
```

Comment:
I think this is a isolation breaker and not generic. I think it made for some specific integration.